### PR TITLE
Adding car endpoint

### PR DESCRIPTION
**Pull request by tonia4ik-s**

### CODE REVIEW

File: Core/Validation/CarCreatingValidation.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    public class CarCreatingValidation : AbstractValidator<CreateCarDTO>
+    {
+        public CarCreatingValidation()
+        {
+            RuleFor(car => car.Model)
+                .NotEmpty()
+                .Matches("^[a-zA-Z0-9 ]*$");
+            RuleFor(car => car.Color)
+                .NotEmpty()
+                .Matches("^[a-zA-Z_.-]*$");
```
Comment (Eugene-pas) [inline Core/Validation/CarCreatingValidation.cs]: Check the correctness of the structure of regular expressions

File: Core/Services/CarService.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            _carRepository = car;
+            _categoryRepository = categoryRepository;
+            _mapper = mapper;
+            _userManager = userManager;
+        }
+        public async Task<CarDTO> AddCarAsync(CreateCarDTO createCarDTO, string userId)
+        {
+            var car = _mapper.Map<Car>(createCarDTO);
+            car.CreationDate = DateTimeOffset.UtcNow;
+            car.Category = await _categoryRepository.GetByIdAsync(createCarDTO.CategoryId);
```
Comment (vladtymo) [inline Core/Services/CarService.cs:39]: You can set the category id only

File: Core/Services/CarService.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            
+            var user = await _userManager.FindByIdAsync(userId);
+            car.User = user ?? throw new HttpException(ErrorMessages.UserNotFound, HttpStatusCode.NotFound);
+            car.User.HasCar = true;
+            
+            await _carRepository.InsertAsync(car);
+            await _carRepository.SaveChangesAsync();
+            return _mapper.Map<CarDTO>(car);
```
Comment (vladtymo) [inline Core/Services/CarService.cs:47]: Does the returned object have the actual id value which was set?

File: Core/Services/CarService.cs (hunk 2 above)
Comment (tonia4ik-s) [reply Core/Services/CarService.cs:39]: I need the whole category object because in DTO I return not categoryId, but it's name.

**Review by vladtymo — APPROVED** [review]

**Review by AndriiChepeliuk — APPROVED** [review]

**Pull request merged**